### ISSUE TITLE
Request for cbDice_func file

### ISSUE THREAD

**Issue by 2448581001** [psweens/VAN-GAN#8]
Dear psweens,
I'm exploring your Project on GitHub. I noticed the cbDice_func file is missing. It seems essential for the project.​
Could you please upload it to the repo? It would help me and other users understand and contribute to the project.​
Thanks for your time. Looking forward to your reply.​
Best regards,​
jerry

**Comment by psweens** [comment]
Sorry Jerry, the cbDice_func is part of my work in progress branches and is not ready for public consumption. If you would like to contribute, please branch from 'main'.

Best,
Paul

**Comment by 2448581001** [comment]
Oops! I'm sorry. I previously thought 2.1.5 was an optimized version. 
When running the main script, I found the following issues and would appreciate it if we could take a look together:
In vangan.py, should reconstruction_loss_I be changed to reconstruction_loss when calculating total_loss_S?
In the downsample function, dropout_rate seems to be undefined.
Thanks for your help and quick response.

**Comment by psweens** [comment]
No problem! I'm currently working on a better more generalisable version which is this branch which should be released in a few months.

Just to double check, you issues are now with release v1.2.1?

**Comment by 2448581001** [comment]
I trained the default version, but this problem might also exist in v1.2.1. And after I changed reconstruction_loss_I to reconstruction_loss, every loss value was nan after the first epoch of training, I wonder if you can give me some suggestions. Thanks again! 

As seen in the vscode, reconstruction_loss_I is not defined.

![Image](https://github.com/user-attachments/assets/20b08803-a09c-44bb-b29c-7895bf07af36)

**Comment by 2448581001** [comment]
![Image](https://github.com/user-attachments/assets/4874db3f-7902-4c44-9678-1e850e91a4bf)

**Comment by psweens** [comment]
Thanks for raising these issues Jerry. I've made the corrections to the code and have double checked that it trains as expected using the synthetic paired dataset from the published paper. The update is given in v.1.2.2.

A previous user has raised that they may be some issues with the synthetic binary masks which can be downloaded the [Cambridge Data Repository](https://www.repository.cam.ac.uk/items/44d4e593-7688-4c1a-b885-c84c2d38c211). The problem-free binary masks can be found [here](https://www.dropbox.com/scl/fi/qh660t6hx93acg1r9olis/LNet.zip?rlkey=4dvpdg6x5pib8ufx6lhgwt90g&dl=0).

**Comment by 2448581001** [comment]
Thanks for all your help, it is really nice of you!!

**Comment by psweens** [comment]
No problem! Happy to help.

**Comment by 2448581001** [comment]
One last question please, how could I use the trained model to segment new images? Is there a separate test file, or some other way.

**Comment by psweens** [comment]
Hi Jerry. Apologies for the delay. Unfortunately, the trained model were not released with the paper.

**Comment by 2448581001** [comment]
Sorry, my expression may be wrong. What I mean is, is there a file like test.py that uses checkpoints to segment new images, because the images segmented by the test.py file I wrote are very different from the results obtained by segmenting the validation set images by the model.

**Comment by 2448581001** [comment]
[test.zip](https://github.com/user-attachments/files/19823629/test.zip)This is the test.py file I wrote. Please help me check it. Thank you ever so much.

**Comment by psweens** [comment]
I suggest using the load_checkpoint functionality within the VANGAN as it sounds like the checkpoints aren't being loaded correctly.

**Comment by 2448581001** [comment]
OK, I'll try it! I haven't noticed that function before. Thanks again.

**Comment by 2448581001** [comment]
It works！
